File: ub/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <unistd.h>
void Greedy(int);




int main(void)
{
    int cents;
    float change;
    //Prompt user for input
    do
    {
        change = get_float("Change Owed: ");

    }
    while (change < 0);
    //turn dollars to cents
    cents = round(change * 100);
    //conver to cahnge
    printf("%d\n", cents);
    Greedy(cents);
  
}
//get the highest change
void Greedy(int amount)
{
    int count = 0;
    do
    {
        //if greater then 25
        if (amount >= 25)
        {
           
            //check if money amount is 0
            if (amount > 0)
            {
                amount -= 25;
                //increase count
                count += 1 ;
               
            }
            
        }
        //is greater then 10 but less then 25
        else if (amount >= 10 && amount < 25)
        {
            //check if money amount is 0
            if (amount > 0)
            {
                amount -= 10;
                count += 1;
               
            }
        }
        //if amount is greater the 5cent but less then 25 cent and 10 cents
        else if (amount >= 5 && amount < 25 && amount < 10)
        {
            //check if money amount is 0
            if (amount > 0)
            {
                amount -= 5;
                //increase count
                count += 1; 
                
            }
        }
        //if greater then 1 cent but less then 25 10 and 5
        else if (amount >= 1  && amount < 25 && amount < 10 && amount < 5)
        {
            //check if money amount is 0
            if (amount > 0)
            {
                amount -= 1;
                //increase count
                count += 1;
               
            }
        }
    }
    while (amount > 0);
    
    printf("Total change is:%d\n", count);
}    

File: ub/pset6/cash/cash.py
from cs50 import get_float
#run main
def main():
    #loop to get input
    while True:
        change = get_float("Change owed: ");
        if change > 0:
            break
    #convert input
    cents = int(change * 100)
    greed(cents)

def greed(amount):
    count = 0
    #loop until get 0
    while amount > 0:
        #25 cent
        if amount >= 25:
            amount -= 25
            count += 1
        elif amount >= 10 and amount < 25:
            amount -= 10;
            count += 1
        elif amount >= 5 and amount < 25 and amount < 10:
            amount -= 5;
            count += 1;
        elif amount >= 1 and amount < 25 and amount < 10 and amount < 5:
            amount -= 1
            count += 1
    #print amount    
    print(count)
    
main()